File: checker/src/dto/proxy.dto.ts
import { z } from "zod";

export const CreateProxyRequestSchema = z.object({
  proxy: z.string().optional(),
  port: z.string().optional(),
  proxyip: z.boolean().optional(),
  ip: z.string().min(1, "IP address is required"),
  latency: z.number().int().optional(),
  asn: z.number().int().nullable().optional(),
  as_organization: z.string().nullable().optional(),
  asOrganization: z.string().nullable().optional(),
  colo: z.string().nullable().optional(),
  country: z.string().nullable().optional(),
  city: z.string().nullable().optional(),
  region: z.string().nullable().optional(),
  postal_code: z.string().nullable().optional(),
  postalCode: z.string().nullable().optional(),
  latitude: z.string().nullable().optional(),
  longitude: z.string().nullable().optional(),
  is_active: z.boolean().optional(),
});

export type CreateProxyRequest = z.infer<typeof CreateProxyRequestSchema>;

export const UpdateProxyRequestSchema = CreateProxyRequestSchema.partial();

export type UpdateProxyRequest = z.infer<typeof UpdateProxyRequestSchema>;

export const ImportProxyItemSchema = CreateProxyRequestSchema;

export type ImportProxyItem = z.infer<typeof ImportProxyItemSchema>;

export const ImportProxyListSchema = z.object({
  proxies: z.array(ImportProxyItemSchema)
});

File: checker-go/internal/delivery/http/dto/proxy_dto.go
package dto

type CreateProxyRequest struct {
	Proxy           *string `json:"proxy"`
	Port            *string `json:"port"`
	ProxyIP         *bool   `json:"proxyip"`
	IP              string  `json:"ip" validate:"required"`
	Latency         *int    `json:"latency"`
	ASN             *int    `json:"asn"`
	ASOrganization  *string `json:"asOrganization"`  // camelCase
	ASOrgSnake      *string `json:"as_organization"` // snake_case fallback
	Colo            *string `json:"colo"`
	Country         *string `json:"country"`
	City            *string `json:"city"`
	Region          *string `json:"region"`
	PostalCode      *string `json:"postalCode"`      // camelCase
	PostalCodeSnake *string `json:"postal_code"`      // snake_case fallback
	Latitude        *string `json:"latitude"`
	Longitude       *string `json:"longitude"`
	IsActive        *bool   `json:"is_active"`
}

type UpdateProxyRequest struct {
	Proxy           *string `json:"proxy"`
	Port            *string `json:"port"`
	ProxyIP         *bool   `json:"proxyip"`
	IP              *string `json:"ip"`
	Latency         *int    `json:"latency"`
	ASN             *int    `json:"asn"`
	ASOrganization  *string `json:"asOrganization"`
	ASOrgSnake      *string `json:"as_organization"`
	Colo            *string `json:"colo"`
	Country         *string `json:"country"`
	City            *string `json:"city"`
	Region          *string `json:"region"`
	PostalCode      *string `json:"postalCode"`
	PostalCodeSnake *string `json:"postal_code"`
	Latitude        *string `json:"latitude"`
	Longitude       *string `json:"longitude"`
	IsActive        *bool   `json:"is_active"`
}

type ImportProxyListRequest struct {
	Proxies []CreateProxyRequest `json:"proxies" validate:"required,dive"`
}

type PublicProxyItem struct {
	Proxy          string  `json:"proxy"`
	Port           string  `json:"port"`
	ProxyIP        bool    `json:"proxyip"`
	IP             string  `json:"ip"`
	Latency        int     `json:"latency"`
	ASN            *int    `json:"asn"`
	ASOrganization *string `json:"asOrganization"` // camelCase matching frontend expectation
	Colo           *string `json:"colo"`
	Country        *string `json:"country"`
	City           *string `json:"city"`
	Region         *string `json:"region"`
	PostalCode     *string `json:"postalCode"`     // camelCase matching frontend expectation
	Latitude       *string `json:"latitude"`
	Longitude      *string `json:"longitude"`
}
